lib/endpoints/class-wp-rest-controller.php

```diff
@@ -3,111 +3,113 @@
 
 abstract class WP_REST_Controller {
 
+	protected $route_base = '';
+	protected $id_regex    = '[\d]+';
+	protected $namespace   = '';
 	/**
 	 * Register the routes for the objects of the controller.
 	 */
 	public function register_routes() {
```

Comment:
Should we break each part of this into separate methods, so you can subclass individual methods to disable and don't need to copy/pasta the internals for `register_routes()`?